glusterd2/commands/volumes/volume-reset.go

```diff
@@ -0,0 +1,119 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerVolResetStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"vol-reset.Store", storeVolume},
+		{"vol-reset.NotifyVolfileChange", notifyVolfileChange},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volname := mux.Vars(r)["volname"]
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotFound)
+		return
+	}
+
+	var req api.VolOptionResetReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	// Delete the option after checking for volopt flags
+	success := false
+	for _, k := range req.Options {
+		if _, ok := volinfo.Options[k]; ok {
+			if !volinfo.Options[k].VOLOPT_FLAG_NEVER_RESET {
```

Comment:
volinfo.Options is `map[string]string`. This usage is wrong. You have to do `xlator.FindOption(k)` and then check for these flag. Expose function `IsNeverReset` similar to `IsSettable` in `$SRC/glusterd2/xlator/options/options.go`.